src/main/java/frc/robot/RobotContainer.java

```diff
@@ -89,4 +75,26 @@ private void configureButtonBindings() {
     public Command getAutonomousCommand() {
         return autonChooser.getSelected();
     }
+
+    void periodic(){
+        tank.forwardComponent = -1 * controller.getLeftY(); // 1 is forward (adjusted from -1 forward)
+        tank.sideComponent = controller.getRightX(); // 1 is right
+
+        if(controller.getBButton()){ //engage the intake arm by pressing B on the controller
```

Comment:
On top of what ethan said, you can implement this logic with callback-based `JoystickButton`s instead of having it in periodic. The `controller.getXButton()` can be rewritten as `controllerXButton.whenPressed(...)` to run a command that toggles `intake_on` when the button is pressed, and `controller.getBButton()` can be rewritten as `controllerBButton.whileHeld(...)` (unless you want it to be toggle-on-press like the X button).

It's a little more complicated since it uses the command system and callbacks, but I think it makes it cleaner (and handles debouncing and errors from toggling in periodic like below).